R/plotPValHist.R

```diff
@@ -31,104 +30,146 @@
+                                                   xAxisTitle           = "P-value",
+                                                   yAxisTitle           = "Count",
+                                                   hideHistogram        = FALSE,
+                                                   showHistogramDensity = FALSE,
+                                                   showLegend           = FALSE,
+                                                   segregateVariablesBy = list("levels"),
+                                                   events               = events)
         } else {
-            for (i in 1:NumSamples) {
-                s <- SampNames[i]
-                MyPVal <- dplyr::filter(P.Val, grepl(s, Levels))
-                cx.data <- subset(MyPVal, select = Pval)
-                var.annot <- subset(MyPVal, select = -c(Pval))
-                Hist_Pval <- canvasXpress::canvasXpress(data                    = cx.data,
-                                                        varAnnot                = var.annot,
-                                                        histogramData           = TRUE,
-                                                        graphType               = "Scatter2D",
-                                                        colors                  = color,
-                                                        title                   = paste(title, "\n", s),
-                                                        xAxisTitle              = "P-value",
-                                                        yAxisTitle              = "Count",
-                                                        hideHistogram           = FALSE,
-                                                        showHistogramDensity    = FALSE,
-                                                        showLegend              = FALSE,
-                                                        segregateVariablesBy    = list("Levels"))
-
-                plotlist[[i]] = Hist_Pval
+            for (i in 1:samples_num) {
+                sample <- sample_names[i]
+                pval_subset <- dplyr::filter(P.Val, grepl(sample, levels))
+                cx.data <- subset(pval_subset, select = pval)
+                var.annot <- subset(pval_subset, select = -c(pval))
+                hist_pval <- canvasXpress::canvasXpress(data                 = cx.data,
+                                                        varAnnot             = var.annot,
+                                                        histogramData        = TRUE,
+                                                        graphType            = "Scatter2D",
+                                                        colors               = color,
+                                                        transparency         = transparency,
+                                                        title                = paste(title, "\n", sample),
```

Comment:
use **glue** instead of **paste**

Reply:
As discussed, with Connie, we need to avoid using glue or external packages as much as possible. Retaining paste for this function,

Reply:
OK